Feature/36 implement tjs north

JeffHeck opened the pull request:
Basic functionality for this cut.

JeffHeck committed:
NOAA Activity mods:
First cut at implementation

JeffHeck committed:
NOAA mods:
Added unit test to test the HTTP GET response conversion

JeffHeck committed:
NOAA mod:
Added another filter to remove the NWWind row in the HTTP response.

Added a test and handling for a null string for the
modifyRawResponse() method.

JeffHeck committed:
Implemented TJs Seattle (minimum functionality at this time)

JeffHeck committed:
Implemented TJs North Sound.
First cut. Basic functionality.

loki980 commented:
Your "onCreate" for the two TJs implementations has a fair amount of copy/paste.  OO programming generally dictates that if you're copy/pasting, you're doing something wrong.  The way to fix this is to create a shared class or function.

What this solves is the ability to reuse code as well as decrease the number of places needed to refactor or fix a bug.  It also cuts down on compiled code size.

I have a laminated copy of this page sitting next to my desk.  =)
http://blog.codinghorror.com/code-smells/

JeffHeck commented:
Agreed. Only rose scented code should be the standard.

JeffHeck committed:
Cleaned up comments. Followed the JavaDoc guide.
Implemented a TJsUtility class for common code for the 2 TJs activities.

JeffHeck commented:
I cleaned up comments. Used the JavaDoc guide.
Checked the JavaDocs that are generated are as expected (and useful).
Refactored the TJs functionality to eliminate duplicate code.

JeffHeck committed:
Updated Gradle version
